contracts/products/BaseProductV2.sol

```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+import "../GovernableInitializable.sol";
+import "../interface/IPolicyManager.sol";
+import "../interface/IRiskManager.sol";
+import "../interface/ITreasury.sol";
```

Comment:
Idk, Is it ok to use non-upgradeable contract imports in an upgradeable contract?

Reply:
its a proxy factory not an upgradeable contract, but either way it depends on your usage. thats why I had to make GovernableInitializable